string_manipulation.rb

```diff
@@ -18,9 +18,9 @@ def reverse_words(my_words)
   i = 0
   while j < my_words.length - 1
     #find the beginning of the word
-    i += 1 until my_words[i] != " "
+    i += 1 while my_words[i] == " "
     #find the end of the word
-    j += 1 until (my_words[j + 1] == " " && my_words[j] != " ") || (j == my_words.length - 1)
+    j += 1 while (my_words[j + 1] != " " && my_words[j] == " ") && (j != my_words.length - 1)

```

Comment:
Make j = i before finding the end of the word i.e. before line 23. Currently if my_words[0] is a space. Then i will increment on line 21 but j won't.
Also, the condition, `while (my_words[j + 1] != " " && my_words[j] == " ")` is interesting. The loop terminates when either my_word[j] is not a space or my_word[j+1] is a space. I understand a word ending when my_word[j] is not a space but not the other case. Think more through this case. I'm curious, are the tests currently passing?